### PR TITLE
Fix #1107: Update activation name

### LINKED ISSUE #1107: Update activation name
Expose a new API for setting the activation name for activations that are not in the state CREATED, REMOVED, or BLOCKED.
The change should be recorded in the history and the audit.

### PR DESCRIPTION
**Pull request by banterCZ**

### CODE REVIEW

File: powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/database/model/entity/ActivationHistoryEntity.java [hunk 1]
```diff
@@ -216,54 +192,32 @@ public void setActivationVersion(Integer version) {
...
-        int hash = 7;
-        hash = 23 * hash + Objects.hashCode(this.activation);
-        hash = 23 * hash + Objects.hashCode(this.activationStatus);
-        hash = 23 * hash + Objects.hashCode(this.eventReason);
-        hash = 23 * hash + Objects.hashCode(this.externalUserId);
-        hash = 23 * hash + Objects.hashCode(this.timestampCreated);
-        hash = 23 * hash + Objects.hashCode(this.activationVersion);
-        return hash;
+        // hashCode should always return the same value even when autogenerated ID has been set; performance penalty is negligible
+        return getClass().hashCode();
```
Comment (banterCZ) [inline powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/database/model/entity/ActivationHistoryEntity.java:206]: Let's not add the new field `applicationName` and return constant as suggested https://vladmihalcea.com/how-to-implement-equals-and-hashcode-using-the-jpa-entity-identifier/
If agreed, I will update our guidelines.

File: powerauth-java-server/src/test/java/io/getlime/security/powerauth/app/server/controller/api/PowerAuthControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+        assertEquals(1, historyEntries.size());
+
+        final ActivationHistoryEntity historyEntry = historyEntries.iterator().next();
+        assertEquals("my iPhone", historyEntry.getActivationName());
+        assertEquals("ACTIVATION_UPDATED", historyEntry.getEventReason());
+        assertEquals("joe-1", historyEntry.getExternalUserId());
+
+        databaseAudit.flush();
+        @SuppressWarnings("unchecked")
+        final List<Tuple> auditEntries = entityManager.createNativeQuery("select * from audit_log", Tuple.class).getResultList();
```
Comment (banterCZ) [inline powerauth-java-server/src/test/java/io/getlime/security/powerauth/app/server/controller/api/PowerAuthControllerTest.java:96]: - [ ] Improve select to get one record only.

**Review by romanstrobl — APPROVED** [review]
Looks OK, I have two minor suggestions.

File: powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/database/model/AdditionalInformation.java [hunk 3]
```diff
@@ -80,6 +80,11 @@ public static class Reason {
          */
         public static final String ACTIVATION_VERSION_CHANGED = "ACTIVATION_VERSION_CHANGED";
 
+        /**
+         * Logged when the activation has been updated, e.g. updated activation name.
+         */
+        public static final String ACTIVATION_UPDATED = "ACTIVATION_UPDATED";
```
[the remark this answers is not in the document]
Comment (banterCZ) [reply powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/database/model/AdditionalInformation.java:86]: It was generic on purpose. I was open to a later extension, but I agree, let's make it more specific.

**Review by zcgandcomp — CHANGES_REQUESTED** [review]
Overall it is fine. I have only one remark - feel free to discuss. 

Question:
Can we name the methods and objects `updateActivationName`? And the as url use  `/activation/name/update`.

Reason:
We are not updating activation we have a specialized method for updating activation name only. So this is an analogy to the naming of method `updateActivationOtp`

**Pull request merged**